### PR TITLE
HPCC-277790 Minor coverity issues

### PR DESCRIPTION
**Pull request by rpastrana**
- Checks all curl set option results
- Frees allocated escapedKQL string

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp [hunk 1]
```diff
@@ -83,27 +83,44 @@ static void requestLogAnalyticsAccessToken(StringBuffer & token, const char * cl
         static constexpr size_t initialBufferSize = 32768; // 2^15
         MemoryBuffer            captureBuffer(initialBufferSize);
         char                    curlErrBuffer[CURL_ERROR_SIZE];
-        curlErrBuffer[0] = '\0';
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:86]: (unless CURL_ERROR_SIZE is zero...)

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp [hunk 2]
```diff
@@ -166,32 +183,52 @@ static void submitKQLQuery(std::string & readBuffer, const char * token, const c
         CURLcode                curlResponseCode;
         OwnedPtrCustomFree<curl_slist, curl_slist_free_all> headers = nullptr;
         char                    curlErrBuffer[CURL_ERROR_SIZE];
-        curlErrBuffer[0] = '\0';
```
Comment (richardkchapman) [inline system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:169]: As above...

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp [hunk 3]
```diff
@@ -166,32 +183,52 @@ static void submitKQLQuery(std::string & readBuffer, const char * token, const c
...
 
         char * encodedKQL = curl_easy_escape(curlHandle, kql, strlen(kql));
```
Comment (richardkchapman) [inline system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:187]: I think the null termination warning was here - and was a false positive.

**Review by richardkchapman — CHANGES_REQUESTED** [review]

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp (hunk 3 above)
Comment (rpastrana) [reply system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:187]: Agreed, the scan might not have access to the CURL_ERROR_SIZE definition.
honestly, I wasn't ready for code review, but glad you jumped on.

**Comment by rpastrana** [comment]
@richardkchapman I had intended to mention to the reviewer that the checks on the curl set opts (and the resulting exceptions) is a bit of an overkill... but it seems reasonable to bail on the request if setting any of those options (some are optional) fails.

**Review by richardkchapman — APPROVED** [review]

**Comment by richardkchapman** [comment]
Merged, but just noticed there's a typo in the commit title. Oh well...

**Pull request merged**